File: DataModule/src/com/mobnetic/coinguardian/model/CurrencySubunitsMap.java
package com.mobnetic.coinguardian.model;

import java.util.LinkedHashMap;

public class CurrencySubunitsMap extends LinkedHashMap<Long, CurrencySubunit> {

	private static final long serialVersionUID = -7219011491064245859L;
	
	public CurrencySubunitsMap(CurrencySubunit... currencySubunits) {
		for(CurrencySubunit currencySubunit : currencySubunits)
			put(currencySubunit.subunitToUnit, currencySubunit);
	}
	
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/CurrencySubunitsMap.kt
package com.aneonex.bitcoinchecker.datamodule.model

class CurrencySubunitsMap(vararg currencySubunits: CurrencySubunit) : LinkedHashMap<Long, CurrencySubunit>() {
    init {
        for (currencySubunit in currencySubunits) put(currencySubunit.subunitToUnit, currencySubunit)
    }
}